backend/src/storage.ts

```diff
@@ -1,28 +1,127 @@
 import * as fs from 'fs';
 import * as fsp from 'fs/promises';
 import * as path from 'path';
+import { Database } from 'sqlite3';
 
 export const STORAGE_FOLDER = process.env['TMT_STORAGE_FOLDER'] || 'storage';
+export const DATABASE_PATH = path.join(STORAGE_FOLDER, 'database.sqlite');
+export const GAME_SERVERS_TABLE = 'gameServers';
+export const PLAYERS_TABLE = 'players';
+export const MATCHES_TABLE = 'matches';
+export const PLAYER_MATCH_STATS_TABLE = 'playerMatchStats';
 
 export const setup = async () => {
 	await fsp.mkdir(STORAGE_FOLDER, {
 		recursive: true,
 	});
+
+	const db = new Database(DATABASE_PATH);
+	db.serialize(() => {
+		db.run(
+			`CREATE TABLE IF NOT EXISTS ${GAME_SERVERS_TABLE} (
+				ip TEXT,
+				port INTEGER,
+				rconPassword TEXT,
+				usedBy TEXT,
+				canBeUsed BOOLEAN,
+				PRIMARY KEY (ip, port)
+			)`,
+			(err) => {
+				if (err) {
+					console.error(err);
+				}
+			}
+		);
+
+		db.run(
+			`CREATE TABLE IF NOT EXISTS ${PLAYERS_TABLE} ( 
+				steamId TEXT PRIMARY KEY, 
+				name TEXT, 
+				tKills INTEGER, 
+				tDeaths INTEGER, 
+				tAssists INTEGER, 
+				tDiff INTEGER, 
+				tHeadshots INTEGER, 
+				tAdr INTEGER)`,
+			(err) => {
+				if (err) {
+					console.error(err);
+				}
+			}
+		);
+
+		db.run(
+			`CREATE TABLE IF NOT EXISTS ${MATCHES_TABLE} ( 
+				matchId TEXT PRIMARY KEY, 
+				teamA TEXT, 
+				teamAScore TEXT, 
+				teamB TEXT, 
+				teamBScore TEXT, 
+				map TEXT, 
+				winner TEXT)`,
+			(err) => {
+				if (err) {
+					console.error(err);
+				}
+			}
+		);
+
+		db.run(
+			`CREATE TABLE IF NOT EXISTS ${PLAYER_MATCH_STATS_TABLE} ( 
+				steamId TEXT REFERENCES ${PLAYERS_TABLE}(steamId), 
+				matchId TEXT REFERENCES ${MATCHES_TABLE}(matchId), 
+				kills INTEGER, 
+				deaths INTEGER, 
+				assists INTEGER, 
+				diff INTEGER, 
+				headshots FLOAT, 
+				adr FLOAT, 
+				CHECK (headshots >= 0 AND headshots <= 100), 
+				PRIMARY KEY (steamId, matchId))`,
+			(err) => {
+				if (err) {
+					console.error(err);
+				}
+			}
+		);
+	});
+	db.close((err) => {
+		if (err) {
+			console.error('Error closing the database:', err.message);
+		}
+	});
 };
 
-export const write = async <T>(fileName: string, content: T) => {
+export const writeJson = async <T>(fileName: string, content: T) => {
 	await fsp.writeFile(path.join(STORAGE_FOLDER, fileName), JSON.stringify(content, null, 4));
 };
 
+export const writeDB = async (query: string) => {
```

Comment:
I don't know if writing queries directly from every function call is very optimized. Maybe a function that takes a table name and content (like writeJson), and gives an error if the content doesn't match the table structure ?

Because right now this isn't really a "write DB" function, but more of a general "query DB" function. All the code that needs to write in the DB will do an `INSERT` anyway, so why not factorize that and make it more integrated by using TypeScript data structures ?